api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -475,6 +475,10 @@ public NestedField asRequired() {
       return new NestedField(false, id, name, type, doc);
     }
 
+    public NestedField withFieldId(int newId) {
```

Comment:
As as a note for other reviewers this is a new Public API but I think it's very safe to include